src/backend/distributed/planner/distributed_planner.c

```diff
@@ -1413,9 +1423,19 @@ AdjustReadIntermediateResultCost(RangeTblEntry *rangeTableEntry, RelOptInfo *rel
 			rowSizeEstimate += 1;
 		}
 
+
 		/* add the cost of parsing a column */
-		rowCost += get_func_cost(inputFunctionId) * cpu_operator_cost;
+#if PG_VERSION_NUM >= 120000
+		add_function_cost(NULL, inputFunctionId, NULL, &funcCost);
+#else
+		funcCost += get_func_cost(inputFunctionId);
+#endif
 	}
+#if PG_VERSION_NUM >= 120000
+	rowCost += funcCost.per_tuple * cpu_operator_cost;
```

Comment:
Two things, `add_function_cost ()` seems to be already multiplying with `cpu_operator_cost`, so we should not do it again.
```add_function_cost()
...
	/* No support function, or it failed, so rely on procost */
	cost->per_tuple += procform->procost * cpu_operator_cost;
```
Also:
Should we need to add the following?
```C
rowCost += funcCost.startup
```

Reply:
I assume `funcCost.startup` is a one-time cost, as opposed to `per_tuple` which would be the per-row portion